repositories/src/main/scala/cromwell/pipeline/datastorage/dao/repository/ProjectConfigurationRepository.scala

```diff
@@ -9,14 +9,23 @@ import scala.concurrent.{ ExecutionContext, Future }
 class ProjectConfigurationRepository(repository: DocumentRepository)(implicit ec: ExecutionContext) {
 
   private def upsertConfiguration(projectConfiguration: ProjectConfiguration): Future[Unit] =
-    repository.upsertOne(projectConfiguration, "projectId", projectConfiguration.projectId.value)
+    repository.upsertOneByTwoParams(
+      projectConfiguration,
+      "projectId",
+      projectConfiguration.projectId.value,
+      "version",
+      projectConfiguration.version.name
+    )
 
   def addConfiguration(projectConfiguration: ProjectConfiguration): Future[Unit] =
     upsertConfiguration(projectConfiguration)
 
   def updateConfiguration(projectConfiguration: ProjectConfiguration): Future[Unit] =
     upsertConfiguration(projectConfiguration)
 
-  def getById(projectId: ProjectId): Future[Option[ProjectConfiguration]] =
-    repository.getByParam("projectId", projectId.value).map(_.headOption)
+  def getByProjectId(projectId: ProjectId): Future[Option[ProjectConfiguration]] =
+    findByProjectId(projectId: ProjectId).map(_.headOption)
+
+  def findByProjectId(projectId: ProjectId): Future[Seq[ProjectConfiguration]] =
+    repository.getByParam("projectId", projectId.value)
```

Comment:
It's really hard to tell the difference between these two methods based on their names. Use something like `getAllByProjectId` and `getLatestByProjectId`.

Reply:
Do you even use `getByProjectId` ?

Reply:
When we try to get one item we are writing 'get', and if we try to get collection we are writing 'find', as i know. But if I wrong then ok. I`ll fix it.

Reply:
Hmm. I've never seen such approach. Usually `find` returns `Option[T]`, `get` returns `T` and `getAll` / `findAll` return `Seq[T]`.

Reply:
Ok, I`ll do my best.

Reply:
I rename this and some other methods.